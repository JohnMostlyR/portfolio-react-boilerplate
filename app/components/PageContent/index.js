/**
 *
 * PageContent
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { theme } from '../../styles/theme';

import Wrapper from './Wrapper';
import ContentWrapper from './ContentWrapper';

import SpeechBubble from '../../components/SpeechBubble';
import PageHeader from '../../components/PageHeader';
import { BASE_LINE_HEIGHT } from '../../styles/typography';

function PageContent(props) {
  let renderThis = (
    <SpeechBubble
      arrowHeight="9vh"
      arrowPosition="bottom-right"
      backgroundColor={theme.speechBubble.backgroundColor}
      color={theme.speechBubble.color}
      makeAppear={false}
      maxWidth="30em"
      padding={`${BASE_LINE_HEIGHT}rem`}
      showArrowBreakpoint="750px"
    >
      {props.content}
    </SpeechBubble>
  );

  if (props.noSpeechBubble) {
    renderThis = props.content;
  }

  return (
    <Wrapper>
      <PageHeader arrowPosition="bottom-left">{props.title}</PageHeader>
      <ContentWrapper>{renderThis}</ContentWrapper>
    </Wrapper>
  );
}

PageContent.propTypes = {
  content: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
  title: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
  noSpeechBubble: PropTypes.bool,
};

export default PageContent;
