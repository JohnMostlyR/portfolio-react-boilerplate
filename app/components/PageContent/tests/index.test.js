import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import PageContent from '../index';

describe('<PageContent />', () => {
  it('Should render and match the snapshot', () => {
    const wrapper = shallow(<PageContent />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('Should adopt the "title" property', () => {
    const wrapper = shallow(<PageContent title="Testing" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('Should adopt the "content" property', () => {
    const wrapper = shallow(<PageContent content="Testing" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('Should adopt the "noSpeechBubble" property', () => {
    const wrapper = shallow(<PageContent content="Testing" noSpeechBubble />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
