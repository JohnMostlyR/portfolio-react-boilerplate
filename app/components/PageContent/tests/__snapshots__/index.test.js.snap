// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageContent /> Should adopt the "content" property 1`] = `
<Wrapper
  animate={true}
>
  <PageHeader
    arrowPosition="bottom-left"
  />
  <ContentWrapper>
    <SpeechBubble
      arrowHeight="9vh"
      arrowPosition="bottom-right"
      backgroundColor="#aedd2c"
      color="#000"
      isGhost={false}
      makeAppear={false}
      maxWidth="30em"
      padding="1.5rem"
      showArrowBreakpoint="750px"
    >
      Testing
    </SpeechBubble>
  </ContentWrapper>
</Wrapper>
`;

exports[`<PageContent /> Should adopt the "noSpeechBubble" property 1`] = `
<Wrapper
  animate={true}
>
  <PageHeader
    arrowPosition="bottom-left"
  />
  <ContentWrapper>
    Testing
  </ContentWrapper>
</Wrapper>
`;

exports[`<PageContent /> Should adopt the "title" property 1`] = `
<Wrapper
  animate={true}
>
  <PageHeader
    arrowPosition="bottom-left"
  >
    Testing
  </PageHeader>
  <ContentWrapper>
    <SpeechBubble
      arrowHeight="9vh"
      arrowPosition="bottom-right"
      backgroundColor="#aedd2c"
      color="#000"
      isGhost={false}
      makeAppear={false}
      maxWidth="30em"
      padding="1.5rem"
      showArrowBreakpoint="750px"
    />
  </ContentWrapper>
</Wrapper>
`;

exports[`<PageContent /> Should render and match the snapshot 1`] = `
<Wrapper
  animate={true}
>
  <PageHeader
    arrowPosition="bottom-left"
  />
  <ContentWrapper>
    <SpeechBubble
      arrowHeight="9vh"
      arrowPosition="bottom-right"
      backgroundColor="#aedd2c"
      color="#000"
      isGhost={false}
      makeAppear={false}
      maxWidth="30em"
      padding="1.5rem"
      showArrowBreakpoint="750px"
    />
  </ContentWrapper>
</Wrapper>
`;
